### PR TITLE
feat: generate file

### PR DESCRIPTION
**Pull request by anhle10051996**

### CODE REVIEW

File: template/generators/component/index.js.hbs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React, {ReactElement} from 'react'
+import {StyleSheet, View, ViewStyle} from 'react-native'
+
+interface IProps {
+  children: ReactElement
+  style: ViewStyle
```
Comment (ngochuyduong) [inline template/generators/component/index.js.hbs]: Should make style optional

File: template/package.json [hunk 2]
```diff
@@ -12,15 +12,18 @@
...
   },
   "dependencies": {
     "@react-native-async-storage/async-storage": "^1.17.10",
     "@react-navigation/native": "^6.0.13",
     "@react-navigation/native-stack": "^6.9.1",
     "@reduxjs/toolkit": "^1.8.6",
     "i18n-js": "^4.1.1",
+    "i18next": "^22.4.9",
     "react": "18.1.0",
+    "react-i18next": "^12.1.4",
```
Comment (ngochuyduong) [inline template/package.json]: Should remove react-i18next, because this is generate file feature

**Review by ngochuyduong — CHANGES_REQUESTED** [review]

File: template/App.tsx [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
+import {Provider} from 'react-redux'
+import MainLayout from './src/MainLayout'
+import {store} from './src/store/store'
 
 interface TextWithDefaultProps extends Text {
   defaultProps?: {
-    allowFontScaling?: boolean;
-    underlineColorAndroid?: 'transparent';
-  };
+    allowFontScaling?: boolean
```
Comment (loc-nguyenthien) [inline template/App.tsx]: What does defaultProp `allowFontScaling: boolean` mean? Should it be `true` or `false`?

File: template/src/locale/I18nConfig.ts [hunk 4]
```diff
@@ -1,28 +1,34 @@
-import I18n from 'i18n-js'
+import i18n from 'i18next'
+import {getI18n, initReactI18next} from 'react-i18next'
```
Comment (loc-nguyenthien) [inline template/src/locale/I18nConfig.ts]: Why do we need to change our `i18n` feature, I think we can move this into a different PR

File: template/src/navigation/StackNavigation.tsx [hunk 5]
```diff
@@ -12,41 +12,63 @@ export const componentMatch = (stackName: string): Element | string => {
...
 
-export const HomeNavigator = (): Element => {
+interface IProps {}
+
+export const HomeNavigator = (props: IProps) => {
   return (
     <Stack.Navigator>
       <Stack.Screen
-        name={RouteKey.HomeScreen}
+        name={'HomeScreen'}
```
Comment (loc-nguyenthien) [inline template/src/navigation/StackNavigation.tsx]: Why are you hard-code this? Please use `RouteKey` for this

File: template/src/navigation/StackNavigation.tsx [hunk 6]
```diff
@@ -12,41 +12,63 @@ export const componentMatch = (stackName: string): Element | string => {
...
   )
 }
 
-export const AuthNavigator = (): Element => {
+export const AuthNavigator = (props: IProps) => {
   return (
     <Stack.Navigator>
       <Stack.Screen
-        name={RouteKey.LoginScreen}
+        name={'LoginScreen'}
```
Comment (loc-nguyenthien) [inline template/src/navigation/StackNavigation.tsx]: Same above

File: template/src/navigation/StackNavigation.tsx [hunk 7]
```diff
@@ -12,41 +12,63 @@ export const componentMatch = (stackName: string): Element | string => {
...
         component={screenMatch(RouteKey.LoginScreen)}
         options={optionsMatch(RouteKey.LoginScreen)}
       />
       <Stack.Screen
-        name={RouteKey.SignUpScreen}
+        name={'SignUpScreen'}
```
Comment (loc-nguyenthien) [inline template/src/navigation/StackNavigation.tsx]: Same above

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: template/App.tsx [hunk 8]
```diff
@@ -1,27 +1,26 @@
...
 
-(Text as unknown as TextWithDefaultProps).defaultProps =
-  (Text as unknown as TextWithDefaultProps).defaultProps || {};
-(Text as unknown as TextWithDefaultProps).defaultProps!.allowFontScaling =
-  false;
-(Text as unknown as TextWithDefaultProps).defaultProps = {
+;(Text as unknown as TextWithDefaultProps).defaultProps =
+  (Text as unknown as TextWithDefaultProps).defaultProps || {}
+;(Text as unknown as TextWithDefaultProps).defaultProps!.allowFontScaling = false
+;(Text as unknown as TextWithDefaultProps).defaultProps = {
```
Comment (ThinhKimVo) [inline template/App.tsx]: check the format again

File: template/src/store/reducers/index.ts [hunk 9]
```diff
@@ -1,11 +1,14 @@
...
+// Reducer Imports
 import user from './user'
 import app from './app'
 import loading from './loading'
+// Reducer Export
 export * from './app'
 export * from './user'
 
 export default combineReducers({
+  // Reducers
```
Comment (ThinhKimVo) [inline template/src/store/reducers/index.ts:27]: could you remove unnecessary comments?

File: template/src/store/selectors/app.ts [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import {IApp} from '../../constants/interface/redux/AppInterface'
 
-const getAppData = (state): IApp => state.app
+const getAppData = (state: any): IApp => state.app
 
-export const getLoadingIndicator = state => getAppData(state).showGlobalIndicator
-export const getAppStackState = (state): string => getAppData(state).appState
+export const getLoadingIndicator = (state: any) => getAppData(state).showGlobalIndicator
+export const getAppStackState = (state: any): string => getAppData(state).appState
```
Comment (ThinhKimVo) [inline template/src/store/selectors/app.ts]: avoid `any`

File: template/src/store/selectors/index.ts [hunk 11]
```diff
@@ -1,2 +1,3 @@
+// Selector
```
Comment (ThinhKimVo) [inline template/src/store/selectors/index.ts:1]: remove

File: template/src/store/selectors/loading.ts [hunk 12]
```diff
@@ -1,4 +1,4 @@
-export const getLoadingSelector = (state, actionTypes: [string]): boolean => {
+export const getLoadingSelector = (state: any, actionTypes: [string]): boolean => {
```
Comment (ThinhKimVo) [inline template/src/store/selectors/loading.ts]: avoid `any`

File: template/generators/component/index.js.hbs (hunk 1 above)
Comment (anhle10051996) [reply template/generators/component/index.js.hbs]: updated

File: template/package.json (hunk 2 above)
Comment (anhle10051996) [reply template/package.json]: updated

File: template/App.tsx (hunk 3 above)
Comment (anhle10051996) [reply template/App.tsx]: updated

File: template/src/locale/I18nConfig.ts (hunk 4 above)
Comment (anhle10051996) [reply template/src/locale/I18nConfig.ts]: updated

File: template/src/navigation/StackNavigation.tsx (hunk 6 above)
Comment (anhle10051996) [reply template/src/navigation/StackNavigation.tsx]: updated

File: template/src/navigation/StackNavigation.tsx (hunk 7 above)
Comment (anhle10051996) [reply template/src/navigation/StackNavigation.tsx]: updated

File: template/src/navigation/StackNavigation.tsx (hunk 5 above)
Comment (anhle10051996) [reply template/src/navigation/StackNavigation.tsx]: updated

**Review by loc-nguyenthien — APPROVED** [review]

**Comment by ThinhKimVo** [comment]
resolve conflict

File: template/src/store/reducers/index.ts (hunk 9 above)
Comment (anhle10051996) [reply template/src/store/reducers/index.ts:27]: comments use for plop. It base on these comment to add some code when you create screen, store by plop script

File: template/src/store/selectors/index.ts (hunk 11 above)
Comment (anhle10051996) [reply template/src/store/selectors/index.ts:1]: same as above

**Review by TranQuangPhi — APPROVED** [review]

**Review by ngochuyduong — APPROVED** [review]
